contracts/withdrawer/EthereumWithdrawer.sol

```diff
@@ -0,0 +1,391 @@
+    Cauldron[] public degenBoxCauldrons;
+
+    mapping(address => bool) public verified;
+
+    constructor(
+        Cauldron[] memory bentoBoxCauldronsV2_,
+        CauldronV1[] memory bentoBoxCauldronsV1_,
+        Cauldron[] memory degenBoxCauldrons_
+    ) {
+        bentoBoxCauldronsV2 = bentoBoxCauldronsV2_;
+        bentoBoxCauldronsV1 = bentoBoxCauldronsV1_;
+        degenBoxCauldrons = degenBoxCauldrons_;
+
+        MIM.approve(address(MIM3POOL), type(uint256).max);
+
+        USDT.safeApprove(address(SWAPROUTER), type(uint256).max);
+        verified[msg.sender] = true;
+    }
+
+    modifier onlyVerified() {
+        require(verified[msg.sender], "Only verified operators");
+        _;
+    }
+
+    function withdraw() public {
+        uint256 length = bentoBoxCauldronsV2.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(bentoBoxCauldronsV2[i].masterContract().feeTo() == address(this), "wrong feeTo");
+
+            bentoBoxCauldronsV2[i].accrue();
+            (, uint256 feesEarned, ) = bentoBoxCauldronsV2[i].accrueInfo();
+            if (feesEarned > (BENTOBOX.toAmount(MIM, BENTOBOX.balanceOf(MIM, address(bentoBoxCauldronsV2[i])), false))) {
+                MIM.transferFrom(MIM_PROVIDER, address(BENTOBOX), feesEarned);
+                BENTOBOX.deposit(MIM, address(BENTOBOX), address(bentoBoxCauldronsV2[i]), feesEarned, 0);
+            }
+
+            bentoBoxCauldronsV2[i].withdrawFees();
+        }
+
+        length = bentoBoxCauldronsV1.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(bentoBoxCauldronsV1[i].masterContract().feeTo() == address(this), "wrong feeTo");
+
+            bentoBoxCauldronsV1[i].accrue();
+            (, uint256 feesEarned) = bentoBoxCauldronsV1[i].accrueInfo();
+            if (feesEarned > (BENTOBOX.toAmount(MIM, BENTOBOX.balanceOf(MIM, address(bentoBoxCauldronsV1[i])), false))) {
+                MIM.transferFrom(MIM_PROVIDER, address(BENTOBOX), feesEarned);
+                BENTOBOX.deposit(MIM, address(BENTOBOX), address(bentoBoxCauldronsV1[i]), feesEarned, 0);
+            }
+            bentoBoxCauldronsV1[i].withdrawFees();
+        }
+
+        length = degenBoxCauldrons.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(degenBoxCauldrons[i].masterContract().feeTo() == address(this), "wrong feeTo");
+
+            degenBoxCauldrons[i].accrue();
+            (, uint256 feesEarned, ) = degenBoxCauldrons[i].accrueInfo();
+            if (feesEarned > (DEGENBOX.toAmount(MIM, DEGENBOX.balanceOf(MIM, address(degenBoxCauldrons[i])), false))) {
+                MIM.transferFrom(MIM_PROVIDER, address(DEGENBOX), feesEarned);
+                DEGENBOX.deposit(MIM, address(DEGENBOX), address(degenBoxCauldrons[i]), feesEarned, 0);
+            }
+            degenBoxCauldrons[i].withdrawFees();
+        }
+
+        BENTOBOX.withdraw(MIM, address(this), address(this), 0, BENTOBOX.balanceOf(MIM, address(this)));
```

Comment:
Looks amazing, I was thinking of whether to do this withdrawing as a public function so that in case anything before messes up we can still get the balance, and make it a public function with no modifier (restrictions)